src/Mono.Android/Mono.Android.targets

```diff
@@ -260,4 +268,29 @@
       <FileWrites Include="$(IntermediateOutputPath)CheckApiCompatibility.stamp" />
     </ItemGroup>
   </Target>
+
+  <Target Name="UpdateExternalDocumentation">
+    <Exec Command="git clean -xdf obj" WorkingDirectory="$(MSBuildThisFileDirectory)" />
+    <PropertyGroup>
+      <_Binlog>UpdateExternalDocumentation-$([System.DateTime]::Now.ToString ("yyyyMMddTHHmmss")).binlog</_Binlog>
+    </PropertyGroup>
+    <Exec
+        Command="msbuild /restore /p:IncludeAndroidJavadoc=True /bl:$(_Binlog)"
+        IgnoreStandardErrorWarningFormat="True"
+        WorkingDirectory="$(MSBuildThisFileDirectory)"
+    />
+    <PropertyGroup>
+      <_Mdoc Condition=" '$(Pkgmdoc)' != '' ">"$(Pkgmdoc)/tools/mdoc.exe"</_Mdoc>
```

Comment:
@jonathanpeppers : any idea how to make this *work*.  Specifically, how to have the `$(Pkgmdoc)` MSBuild variable, set because of `%(PackageReference.GeneratePathProperty)` on the `mdoc` package, *actually exist* in the "outer" build?

Or is there some way to have the `UpdateExternalDocumentation` run as part of the "inner" build when invoking the target from the command-line?

```
msbuild /t:UpdateExternalDocumentation src/Mono.Android/Mono.Android.csproj
```

Reply:
Unless you want to do this during `Build`, I think you'll have to make 2 targets for this to work...

Didn't test, but something like:
```xml
<Target Name="UpdateExternalDocumentation">
  <ItemGroup>
    <_UEDTargetFramework Include="$(TargetFrameworks)" />
  </ItemGroup>
  <MSBuild Projects="$(MSBuildThisProjectFile)" Targets="_UpdateExternalDocumentation" Properties="TargetFramework=%(_UEDTargetFramework.Identity)" />
 </Target>
 <Target Name="_UpdateExternalDocumentation">
   <!--Do the actual work here-->
 </Target>
 ```
 
 The "outer" one will need to call the `<MSBuild/>` task (batched), passing each value in `$(TargetFrameworks)` as `$(TargetFramework)`.

Reply:
The inner target `_UpdateExternalDocumentation` might also need `DependsOnTargets="ResolveAssemblyReferences"` (or some target that populates `$(Pkg*)`).

Reply:
Got it working.  The inner target does *not* need any dependencies.  However:

1. it can't use be `<MSBuild Projects="$(MSBuildThisProjectFile)"/>`.  It needs to use the `.csproj` file, e.g. `<MSBuild Projects="$(MSBuildThisFileDirectory)Mono.Android.csproj" />.
2. I only want to run *one* inner build, so there's no need to have `@(_UEDTargetFramework)` or use `%(_UEDTargetFramework.Identity)`; just using `Properties="TargetFramework=monoandroid10"` suffices.
3. No `DependsOnTargets` is needed.

This in turn means that PR #5491 doesn't need to "worry about" installing the `mdoc` NuGet package into a top-level `xamarin-android/packages` folder, as we'll properly resolve the `mdoc` NuGet installation location via `$(Pkgmdoc)`.